src/NuGet.Core/NuGet.PackageManagement/BuildIntegration/DependencyGraphRestoreUtility.cs

```diff
@@ -218,11 +218,15 @@ public static async Task<PackageSpec> GetProjectSpec(IDependencyGraphProject pro
         {
             var dgSpec = new DependencyGraphSpec();
 
-            var projects = (await solutionManager.GetNuGetProjectsAsync()).OfType<IDependencyGraphProject>();
+            var stringComparer = PathUtility.GetStringComparerBasedOnOS();
 
-            foreach (var project in projects)
+            var uniqueProjectDependencies = new HashSet<string>(stringComparer);
+
+            var projects = ((await solutionManager.GetNuGetProjectsAsync()).OfType<IDependencyGraphProject>()).ToList();
```

Comment:
How this behave in the case in which we are still waiting for a nomination? 

Didn't we want to distinguish between the NU1105 (the project is not in the solution & you have never restored) and NU110, the project-system is slow.

Reply:
If there is a persisted DGSpec then it will consume it and continue with the restore. And when actual nomination happens for that project, then restore will either be NoOp (if there was no change in the project) or restore will consume the new changes. So IMO this experience is better than showing bunch of errors and the final restore being successful.